framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -256,198 +272,86 @@ private void viewpointAdaptExecutable(
         // Then copy annotations from the viewpoint adapted type to methodType, if that annotation
         // is not on a type that was substituted for a type variable.
 
-        viewpointAdaptToContext(context, viewpointAdaptedType);
+        StringToJavaExpression stringToJavaExpr;
+        if (tree instanceof MethodInvocationTree) {
+            stringToJavaExpr =
+                    stringExpr ->
+                            StringToJavaExpression.atMethodInvocation(
+                                    stringExpr, (MethodInvocationTree) tree, factory.getChecker());
+        } else if (tree instanceof NewClassTree) {
+            stringToJavaExpr =
+                    stringExpr ->
+                            StringToJavaExpression.atConstructorInvocation(
+                                    stringExpr, (NewClassTree) tree, factory.getChecker());
+        } else {
+            throw new BugInCF("Unexpected tree: %s kind: %s", tree, tree.getKind());
+        }
+        convertAnnotatedTypeMirror(stringToJavaExpr, viewpointAdaptedType);
         this.viewpointAdaptedCopier.visit(viewpointAdaptedType, methodType);
     }
 
     /**
-     * Converts method or constructor arguments from Trees to JavaExpressions, accounting for
-     * varargs.
+     * Viewpoint-adapts the Java expressions in annotations written on a field declaration for the
+     * use at {@code fieldAccess}.
      *
-     * @param tree invocation of the method or constructor
-     * @param methodType type of the method or constructor
-     * @param argTrees the arguments to the method or constructor; subexpressions of {@code tree}
-     * @return the arguments, as JavaExpressions
+     * @param fieldAccess a field access
+     * @param type its type; is side-effected by this method
```

Comment:
This comment looks wrong.  `type` is not the type of the field access, but the declared type of the field.  Please review the formal parameter descriptions throughout.

Reply:
This comment is correct.  It's type of the field access, but the annotations are those written at the declaration of the field.  For example:
```
class MyClass<T> {
  Map map = ..;
  @KeyFor("this.map") List<T> field;
}

void method(MyClass<String> myClass) {
  myClass.field;
}
```
The declared type of `field` is `@KeyFor("this.map) List<T>`.  
For the call to `atFieldAccess`, the value of `type` is `@KeyFor("this.map") List<String>`.

I'm not sure how to make this clear in the Javadoc.